fe/src/main/java/org/apache/doris/catalog/TableProperty.java

```diff
@@ -0,0 +1,94 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog;
+
+import org.apache.doris.clone.DynamicPartitionScheduler;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.common.util.TimeUtils;
+import org.json.JSONObject;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class TableProperty implements Writable {
+    private Map<String, String> properties = new HashMap<>();
+
+    private static final String DYNAMIC_PARTITION_PROPERTY_PREFIX = "dynamic_partition";
+    private DynamicPartitionProperty dynamicPartitionProperty;
+
+    public TableProperty() {}
+
+    public TableProperty(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public DynamicPartitionProperty getDynamicPartitionProperty() {
+        return dynamicPartitionProperty;
+    }
+
+    public void modifyTableProperties(Map<String, String> modifyProperties) {
+        properties.putAll(modifyProperties);
+        this.dynamicPartitionProperty = buildDynamicProperty(properties);
+    }
+
+    private DynamicPartitionProperty buildDynamicProperty(Map<String, String> properties) {
+        HashMap<String, String> dynamicPartitionProperties = new HashMap<>();
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            if (entry.getKey().startsWith(DYNAMIC_PARTITION_PROPERTY_PREFIX)) {
+                dynamicPartitionProperties.put(entry.getKey(), entry.getValue());
+                DynamicPartitionScheduler.setLastUpdateTime(TimeUtils.getCurrentFormatTime());
+            }
+        }
+        return new DynamicPartitionProperty(dynamicPartitionProperties);
+    }
+
+    public static TableProperty read(DataInput in) throws IOException {
+        TableProperty info = new TableProperty();
+        info.readFields(in);
+        return info;
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        JSONObject jsonObject = new JSONObject();
```

Comment:
There is a constructor of `JSONObject` you can use:
`JSONObject jsonObject = new JSONObject(properties);`

Reply:
Done